core/src/com/unciv/logic/city/managers/CityInfoConquestFunctions.kt

```diff
@@ -315,6 +315,13 @@ class CityInfoConquestFunctions(val city: City){
 
             if (civ.gameInfo.isReligionEnabled()) religion.removeUnknownPantheons()
 
+            if (newCiv.getMatchingUniques(UniqueType.MayNotAnnexCities).any()) {
```

Comment:
I was not aware of hasUnique. As for Conditionals, I was thinking of implementing them sometime later (as I don't really see where they might be used) though I suppose it would be better to do it all at once.

As for AI - well, I completely forgot about that. My bad.